lib/rspec/matchers/built_in/compound.rb

```diff
@@ -101,6 +123,11 @@ def matcher_supports_block_expectations?(matcher)
           false
         end
 
+        def matcher_list_for(matcher)
```

Comment:
One other thing about `diffable?`: it's an optional part of the matcher protocol.  That means that some matchers may not implement it.  Thus, we need to check `respond_to?(:diffable?)` before sending that message, like it's done here:

https://github.com/rspec/rspec-expectations/blob/45ff9faa26ea912d6b13d4044e5691b894d9fcc5/lib/rspec/expectations/handler.rb#L37